frontend/src/app/matching/page.tsx

```diff
@@ -1,91 +1,111 @@
 'use client';
 
 import {
-  FormControl,
-  FormLabel,
-  Select,
-  Flex,
+  Text,
   Button,
   Box,
+  useColorModeValue,
+  Center,
+  VStack,
 } from '@chakra-ui/react';
-import { Difficulty } from '@/constants/difficulty';
+import { QuestionComplexity } from '@/types/question';
 import {
+  DISCONNECT,
   REQ_FIND_PAIR,
+  RES_CANNOT_FIND_PAIR,
   RES_FIND_PAIR,
+  RES_FOUND_PAIR,
 } from '@/constants/socket';
 import { useEffect, useState } from 'react';
 import io from 'socket.io-client';
+import 'dotenv/config';
+import QuestionRangeSlider from '../../components/slider/QuestionRangeSlider';
 
-require("dotenv").config();
+const socket = io(process.env.MATCHING_PORT || 'http://localhost:6006', {
+  autoConnect: false,
+});
 
-const socket = io(process.env.MATCHING_PORT || 'http://localhost:6006');
+function Page() {
+  const [lowerBoundDifficulty, setLowerBoundDifficulty] =
+    useState<QuestionComplexity>(QuestionComplexity.EASY);
 
-export default function Page() {
+  const [upperBoundDifficulty, setUpperBoundDifficulty] =
+    useState<QuestionComplexity>(QuestionComplexity.HARD);
 
-  const modalTitle = 'Choose Matching Difficulty';
-  const placeholder = 'Choose difficulty';
-
-  const [difficulty, setDifficulty] = useState<Difficulty>(
-    Difficulty.EASY,
-  );
-
-  const changeDifficulty = (e) =>
-    setDifficulty(
-      Difficulty[
-        e.target.value as keyof typeof Difficulty
-      ],
-    );
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isConnected, setIsConnected] = useState(false);
 
   useEffect(() => {
     // Listen for 'message' event from the server
-    const socket = io(process.env.MATCHING_PORT || 'http://localhost:6006');
+    const onConnect = () => {
+      // console.log('Connected to server.');
+      setIsConnected(true);
+    };
 
-    socket.on('connection', () => {
-      console.log("Connected to server.");
-    });
+    socket.on('connect', onConnect);
 
+    socket.connect();
 
     return (): void => {
-      socket.disconnect();
+      socket.off('connect', onConnect);
```

Comment:
useEffect changed to remove listener from connect event rather than directly disconnect